### PR TITLE
Start/stop animation preview in Game Lab on focus/blur

### PR DESCRIPTION
**Pull request by bencodeorg**
In the animation picker in Game Lab, the animation previews on hover and stops previewing when your cursor leaves the animation. This is an accessibility no-no, as these previews are then unavailable to folks using tab navigation.

This PR updates the animation picker so that the preview plays if you tab into the animation, and stops playing when you tab out (see video below).

The linter rule is strictly still violated with my implementation, so I added no-op handlers and comments to describe why they're there. I also turned off the linter rule briefly and saw that the violation is no longer given.

<details>
  <summary><b>Video of updated experience</b></summary>
  <video src='https://github.com/user-attachments/assets/8491afc6-45a0-4eae-b462-6216bd98e5c0' />
</details>

## Links

- jira ticket: [A11Y-340](https://codedotorg.atlassian.net/browse/A11Y-340)

## Testing story

Tested manually that animations (eg, the flower animation in the video above) and static images (eg, the "blocks" image above) were appropriately turned on/off and unaffected, respectively, via tab navigation.

### CODE REVIEW

File: apps/src/p5lab/AnimationPicker/AnimationPreview.jsx [hunk 1]
```diff
@@ -48,6 +49,24 @@ export default class AnimationPreview extends React.Component {
     }
   }
 
+  componentDidUpdate(prevProps) {
+    // Only start/stop animations via focus if there is no playBehavior set.
+    // The playBehavior props sets the animation to always play or never play.
```
Comment (fisher-alice) [inline apps/src/p5lab/AnimationPicker/AnimationPreview.jsx]: Nit: Wasn't sure what this meant at first. Maybe update to something like:
'...sets the animation to 'ALWAYS_PLAY' or 'NEVER_PLAY' which determines if animation preview repeatedly advances to next frame.'

File: apps/src/p5lab/AnimationPicker/AnimationPreview.jsx [hunk 2]
```diff
@@ -171,6 +190,12 @@ export default class AnimationPreview extends React.Component {
     }
 
     return (
+      /*
+        The behavior managed by the onMouseOver and onMouseOut handlers is replicated on focus and blur
+        in the componentDidUpdate lifecycle method for accessibility purposes.
+        This div is never actually focused, so using native onFocus and onBlur handlers doesn't work without substantial changes. 
+        We include no-op onFocus and onBlur handlers to suppress a11y linter errors.
+      */
```
Comment (fisher-alice) [inline apps/src/p5lab/AnimationPicker/AnimationPreview.jsx:199]: Thanks for the comment!
Just curious, which div is actually focused? AnimationListItem?

**Review by fisher-alice — APPROVED** [review]
Pulled locally and works with keyboard navigation - nice! Thanks for the explanations. Just left a question and nit comment.

File: apps/src/p5lab/AnimationPicker/AnimationPreview.jsx (hunk 2 above)
Comment (bencodeorg) [reply apps/src/p5lab/AnimationPicker/AnimationPreview.jsx:199]: Yep! It's the top-level `button` that wraps everything else in `AnimationListItem`

File: apps/src/p5lab/AnimationPicker/AnimationPreview.jsx (hunk 1 above)
Comment (bencodeorg) [reply apps/src/p5lab/AnimationPicker/AnimationPreview.jsx]: Good call, updated!

File: apps/src/p5lab/AnimationPicker/AnimationPreview.jsx [hunk 3]
```diff
@@ -48,6 +49,25 @@ export default class AnimationPreview extends React.Component {
     }
   }
 
+  componentDidUpdate(prevProps) {
```
Comment (bencodeorg) [inline apps/src/p5lab/AnimationPicker/AnimationPreview.jsx:52]: For posterity, there's some similar stop/start animation management in the `UNSAFE_componentWillReceiveProps` that I do not fully understand. This new code and the existing code seem not to conflict in meaningful ways as far as I can tell, but seems plausible some simplification/clean up could be done here.

Comment (fisher-alice) [reply apps/src/p5lab/AnimationPicker/AnimationPreview.jsx:52]: Thanks! Just logging that https://github.com/code-dot-org/code-dot-org/pull/42220 updated the deprecated lifecycle methods `componentWillReceiveProps`and `componentWillMount` with the prefix `UNSAFE_`.

**Pull request merged**